crates/matrix-sdk/src/client/mod.rs

```diff
@@ -1510,16 +1544,41 @@ impl Client {
     /// let msc_x = unstable_features.get("msc_x").unwrap_or(&false);
     /// # anyhow::Ok(()) };
     /// ```
-    pub async fn unstable_features(&self) -> HttpResult<&BTreeMap<String, bool>> {
-        let unstable_features = self
-            .inner
-            .unstable_features
-            .get_or_try_init(|| self.request_unstable_features())
-            .await?;
+    pub async fn unstable_features(&self) -> HttpResult<BTreeMap<String, bool>> {
+        if let Some(unstable_features) = self.inner.unstable_features.read().await.as_ref() {
```

Comment:
This is almost the same as `server_version()` I think we can remove a bunch of code here, especially if those two pieces of data live behind the same lock.